File: Source/JavaScript/model/elements/panels/StackPanel.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { Panel } from '../Panel';
import { Orientation } from '../../common';

/**
 * Lays its children out in a single line.
 */
export interface StackPanel extends Panel {
    orientation: Orientation;
    spacing: number;
}

export const StackPanelPropertyNames: (keyof StackPanel)[] = ['orientation', 'spacing'];

File: Source/DotNET/Model/Elements/Panels/StackPanel.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Scene.Model.Common;

namespace Cratis.Scene.Model.Elements.Panels;

/// <summary>
/// Lays its children out in a single line.
/// </summary>
public record StackPanel : Panel
{
    /// <summary>
    /// Gets the direction the line runs in.
    /// </summary>
    public Orientation Orientation { get; init; } = Orientation.Vertical;

    /// <summary>
    /// Gets the space left between one child and the next.
    /// </summary>
    public double Spacing { get; init; }
}
